spec/requests/custom_resources_create_spec.rb

```diff
@@ -0,0 +1,696 @@
+
+      let!(:json) { Map JSON.parse response.body }
+      let!(:attributes) { json.data.attributes }
+
+      it 'has a list of attributes' do
+        expect(attributes).to include(
+          'contributors',
+          'coverageStatement',
+          'customCoverages',
+          'customEmbargoPeriod',
+          'description',
+          'identifiers',
+          'isPeerReviewed',
+          'isSelected',
+          'managedCoverages',
+          'managedEmbargoPeriod',
+          'name',
+          'packageId',
+          'packageName',
+          'providerId',
+          'providerName',
+          'publicationType',
+          'publisherName',
+          'subjects',
+          'titleId',
+          'url',
+          'vendorId',
+          'vendorName',
+          'visibilityData'
+        )
+      end
+      it 'has the new name' do
+        expect(json.data.attributes.name).to eq('New Custom Title Testing')
+      end
+
+      it 'has the new publication type' do
+        expect(json.data.attributes.publicationType).to eq('Book')
+      end
+
+      it 'has the newly generated id' do
+        expect(json.data.attributes.titleId).to equal(17_092_181)
+      end
+
+      it 'has the provider id' do
+        expect(json.data.attributes.providerId).to equal(123_355)
+      end
+
+      it 'has the composite id' do
+        expect(json.data.id).to eq('123355-2843712-17092181')
+      end
+    end
+
+    describe 'with an existing name' do
+      let(:params) do
+        {
+          'data' => {
+            'type' => 'resources',
+            'attributes' => {
+              'name' => 'New Custom Title Testing',
+              'publicationType' => 'Book',
+              'packageId' => 2_843_712
+            }
+          }
+        }
+      end
+
+      before do
+        VCR.use_cassette('post-custom-resource-existing-name') do
+          post '/eholdings/resources',
+               params: params, as: :json, headers: create_headers
+        end
+      end
+
+      let!(:json) { Map JSON.parse response.body }
+
+      it 'responds with bad request' do
+        expect(response).to have_http_status(400)
+        expect(json.errors.first.title).to eql('Custom Title with the provided name already exists')
+      end
+    end
+
+    describe 'with missing name' do
+      let(:params) do
+        {
+          'data' => {
+            'type' => 'resources',
+            'attributes' => {
+              'publicationType' => 'Book',
+              'packageId' => 2_843_712
+            }
+          }
+        }
+      end
+
+      before do
+        VCR.use_cassette('post-custom-resource-missing-name') do
+          post '/eholdings/resources',
+               params: params, as: :json, headers: create_headers
+        end
+      end
+
+      let!(:json) { Map JSON.parse response.body }
+
+      it 'returns an error status' do
+        expect(response).to have_http_status(422)
+        expect(json.errors.first.title).to eql('Invalid titleName')
```

Comment:
can we have a message here like "titleName cannot be blank" to be more explicit?

Reply:
Below is the full response that is returned - the details indicates that the title is blank . This is consistent with what was done for custom package create .
{
    "errors": [
        {
            "title": "Invalid titleName",
            "detail": "Titlename can't be blank",
            "source": {}
        }
    ],
    "jsonapi": {
        "version": "1.0"
    }
}